pkg/eks/nodegroup_service.go

```diff
@@ -65,8 +65,14 @@ func (m *NodeGroupService) NewAWSSelectorSession(provider api.ClusterProvider) *
 func (m *NodeGroupService) Normalize(nodePools []api.NodePool, clusterMeta *api.ClusterMeta) error {
 	for _, np := range nodePools {
 		switch ng := np.(type) {
+		case *api.ManagedNodeGroup:
+			if ng.AMIFamily != api.NodeImageFamilyAmazonLinux2 && !api.IsAMI(ng.AMI) {
+				if err := ResolveAMI(m.Provider, clusterMeta.Version, np); err != nil {
+					return err
+				}
+			}
```

Comment:
again slightly confused on why we don't have to `if` on bottlerocket?

Reply:
Only Bottlerocket requires AMI resolution in managed nodegroups. We plan on adding support for Ubuntu and Windows later, so instead of explicitly stating all of them, I chose to compare it with the one AMI family that doesn't require AMI resolution. I agree that this condition can be made clearer, so I'll work on something.

Reply:
I have pushed a change that should make the condition more clear.